neural/predict_with_neural.py

```diff
@@ -0,0 +1,217 @@
+                to_gpu, gpu_index, num_epochs=1):
+    fold_num = 0
+    q_predictions = {}
+    for question in q_fold_config_map:
+        mconfig, options, __ = q_fold_config_map[question]
+        options['num_epochs'] = num_epochs  # override number of epochs using user specified value
+
+        # update options fold num to the current fold
+        options['fold_num'] = fold_num
+        data_partition = q_docpartitions[question][fold_num]
+
+        results_path = os.path.join(results_dir, 'question_{}'.format(question), 'fold_{}'.format(fold_num))
+        results_wrk_dir = create_directory(results_path)
+
+        q_predictions[question] = predict_neural_discern(data_partition, bertmodel, mconfig, options,
+                                                         results_wrk_dir, sents_embed_dir,
+                                                         state_dict_dir=q_state_dict_path_map[question], to_gpu=to_gpu,
+                                                         gpu_index=gpu_index)
+    return q_predictions
+
+
+def retrieve_page_from_internet(url: str):
+    res = requests.get(url)
+    html_page = res.content.decode("utf-8")
+    return html_page
+
+
+def create_prediction_qdoc_partitions(questions: List[int], fold_num: int):
+    q_docpartitions = {}
+    for q in questions:
+        q_docpartitions[q] = {}
+        q_docpartitions[q][fold_num] = {'train': [], 'validation': [], 'test': [0]}  # output of np.vectorize() ?!
+    return q_docpartitions
+
+
+def build_DataDictProcessor(data_dict, vocab_path: str, processor_config: Dict):
+    # from the first notebook
+
+    processor = DataDictProcessor(processor_config)
+    processor.generate_articles_repr(data_dict)
+    return processor
+
+
+def embed_sentences(docs_data_tensor, sents_embed_path, bertmodel, bert_config, gpu_index):
+    # from the second notebook
+
+    bertembeder = BertEmbedder(bertmodel, bert_config)
+    fdtype = torch.float32
+
+    # generate and dump bert embedding for the tokens inside the specificed embedding directory
+    bert_proc_docs = generate_sents_embeds_from_docs(docs_data_tensor, bertembeder, sents_embed_path, fdtype,
+                                                     gpu_index=gpu_index)
+    ReaderWriter.dump_data(bert_proc_docs, os.path.join(sents_embed_path, 'bert_proc_docs.pkl'))
+
+    return sents_embed_path
+
+
+def biobert_predict(data_dict: dict):
+    """
+    Make an autoDiscern prediction for an article data_dict using the HEA BioBERT model. Includes all of the data
+    preprocessing steps as were applied for the training of the HEA BioBERT model.
+    Args:
+        data_dict: dictionary of {id: sib-dict}, with sub-dictionary with keys ['url', 'content', 'id', 'responses']
+    Returns: autodiscern predictions for the article.
+    """
+
+    base_dir = BASE_DIR
+    to_gpu = True
+    gpu_index = 3
+    fold_num = 0
+    working_dir = 'predict'
+    experiment_dir = '2019-10-08_14-54-50'
```

Comment:
i am tagging here just to double check the `experiment folder` trained for BioBert

Reply:
Yes, thank you, fixed! :grimace: (It was a BioBERT experiment dir, but not the most up-to-date one)